packages/smithy-http/src/smithy_http/aio/crt.py

```diff
@@ -128,21 +133,21 @@ def on_body(self, chunk: bytes, **kwargs: Any) -> None:  # pragma: crt-callback
             self._received_chunks.append(chunk)
 
     async def next(self) -> bytes:
-        if self._stream is None:
+        if self._completion_future is None:
             raise SmithyHTTPException("Stream not set")
 
         # TODO: update backpressure window once CRT supports it
         if self._received_chunks:
             return self._received_chunks.popleft()
-        elif self._stream.completion_future.done():
+        elif self._completion_future.done():
```

Comment:
Note that async futures return immediately on done, whereas concurrent futures can wait... forever